--- galera-25.3.5-src.old/chromium/build_config.h	2014-03-27 15:49:30.000000000 +0000
+++ galera-25.3.5-src/chromium/build_config.h	2014-11-27 21:57:59.377756193 +0000
@@ -106,6 +106,11 @@
 #define ARCH_CPU_ARMEL 1
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_LITTLE_ENDIAN 1
+#elif defined(__aarch64__)
+#define ARCH_CPU_ARM_FAMILY 1
+#define ARCH_CPU_ARMEL 1
+#define ARCH_CPU_64_BITS 1
+#define ARCH_CPU_LITTLE_ENDIAN 1
 #elif defined(__pnacl__)
 #define ARCH_CPU_32_BITS 1
 #elif defined(__MIPSEL__)
